File: server/src/main/kotlin/ovpn/closedadmin/server/business/vpn/controller/MyVpnController.kt
package ovpn.closedadmin.server.business.vpn.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ovpn.closedadmin.server.business.account.vo.VpnUser
import ovpn.closedadmin.server.business.admin.vo.Admin
import ovpn.closedadmin.server.business.vpn.usecase.GetOvpnFile
import ovpn.closedadmin.server.common.dto.CommonResponse
import java.util.UUID

@RestController
@RequestMapping("/myvpn")
class MyVpnController {

    @Autowired
    lateinit var getOvpnFile: GetOvpnFile

    @PostMapping("/download")
    fun downloadOvpnFileAdmin(
        @RequestHeader(name = "VpnAuthentication") vpnUser: VpnUser,
        ): CommonResponse<String> {
            var file: String = getOvpnFile.run(vpnUser.uid.toString())
            return CommonResponse(file)
    }
}

File: web/src/business/vpn/controller/MyVpnController.ts
import {UUID} from "crypto";
import {MyVpnHttpUtil} from "../../../common/util/MyVpnHttpUtil";

const myVpnHttpUtil = new MyVpnHttpUtil();

export class MyVpnController {
    async getOvpnFile(): Promise<string> {
        const response = await myVpnHttpUtil.post<string, null>('/myvpn/download', null)
        return response.response!;
    }
}
